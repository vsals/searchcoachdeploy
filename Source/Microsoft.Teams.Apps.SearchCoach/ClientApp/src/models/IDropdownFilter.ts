// <copyright file="IDropdownFilter.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/**
* Interface to work with drop-down list item values.
*/
export interface IDropdownListItem {
    name: string;
    id: string;
}

/**
* Interface to work with drop-down list selected item value.
*/
export interface ISelectedDropdownItem {
    header: string,
    key: string,
}