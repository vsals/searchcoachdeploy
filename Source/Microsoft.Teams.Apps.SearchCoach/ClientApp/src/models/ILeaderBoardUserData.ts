// <copyright file="ILeaderBoardUserData.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// Interface to deal with user response data for leader-board tab.
export interface ILeaderBoardUserData {
    userName: string;
    rightAnswers: number;
    questionsAttempted: number;
}