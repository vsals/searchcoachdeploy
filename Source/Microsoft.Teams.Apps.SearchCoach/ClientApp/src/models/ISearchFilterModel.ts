// <copyright file="ISearchFilterModel.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/** This interface is used for maintaining the state of domain check-box values. */
export interface ISearchFilterModel {
    // searched text by user.
    searchText: string;
    // selected domains using domain filter
    domains?: Array<string>;
    // selected freshness value using freshness filter
    freshness: string;
    // selected country-code using location filter
    market: string;
}