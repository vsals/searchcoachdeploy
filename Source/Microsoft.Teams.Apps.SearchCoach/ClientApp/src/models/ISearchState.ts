// <copyright file="ISearchState.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/**
* Search state interface.
*/
export interface ISearchState {
    // search text string entered by user.
    searchText: string;
    // country selected string by user.
    selectedCountry: string;
}