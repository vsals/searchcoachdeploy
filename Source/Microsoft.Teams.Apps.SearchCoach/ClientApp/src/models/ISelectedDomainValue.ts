// <copyright file="ISelectedDomainValue.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// This interface is used for maintaining the state of domain check-box values.
export interface ISelectedDomainValue {
    // unique value to identity each item.
    key: string;
    // name of the check-box item.
    title: string;
    // explains whether the check-box is checked or unchecked.
    isChecked: boolean;
}