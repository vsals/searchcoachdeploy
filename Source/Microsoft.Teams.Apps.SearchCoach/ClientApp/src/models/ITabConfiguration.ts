// <copyright file="ITabConfiguration.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// Interface to deal with mapping of teams configurable tab.
export interface ITabConfiguration {
    tabId?: string,
    teamId: string,
    groupId: string,
}