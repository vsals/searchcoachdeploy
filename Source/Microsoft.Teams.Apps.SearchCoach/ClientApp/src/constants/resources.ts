// <copyright file="resources.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import { IDropdownListItem } from "models/IDropdownFilter";

export default class Resources {

    // themes
    public static readonly body: string = "body";
    public static readonly theme: string = "theme";
    public static readonly default: string = "default";
    public static readonly light: string = "light";
    public static readonly dark: string = "dark";
    public static readonly contrast: string = "contrast";

    // screen size
    public static readonly screenWidthLarge: number = 1200;
    public static readonly screenWidthSmall: number = 1000;

    // Countries list to be shown in location filter
    public static readonly countries: IDropdownListItem[] = [
        { name: "No Filter", id: "nf" } as IDropdownListItem,
        { name: "United States", id: "en-US" } as IDropdownListItem,
        { name: "Japan", id: "ja-JP" } as IDropdownListItem,
        { name: "France", id: "fr-FR" } as IDropdownListItem,
        { name: "Germany", id: "de-DE" } as IDropdownListItem,
        { name: "Italy", id: "it-IT" } as IDropdownListItem,
        { name: "South Korea", id: "ko-KR" } as IDropdownListItem,
        { name: "Russia", id: "ru-RU" } as IDropdownListItem
    ];

    // Domains list to be shown in domain filter
    public static readonly domains: IDropdownListItem[] = [
        { name: ".com", id: "1" } as IDropdownListItem,
        { name: ".org", id: "2" } as IDropdownListItem,
        { name: ".edu", id: "3" } as IDropdownListItem,
        { name: ".net", id: "4" } as IDropdownListItem,
        { name: ".gov", id: "5" } as IDropdownListItem,
        { name: ".mil", id: "6" } as IDropdownListItem,
    ];

    // Filter buttons
    public static readonly operatorText: string = "operators";
    public static readonly locationText: string = "location";
    public static readonly domainText: string = "alldomains";
    public static readonly anytimeText: string = "anytime";

    // Marking unknown category or country
    public static readonly unknownText: string = "Unknown";

    // Operator filters
    public static readonly andOperatorText: string = "AND";
    public static readonly orOperatorText: string = "OR";
    public static readonly notOperatorText: string = "NOT";

    // Freshness values
    public static readonly freshnessDayText: string = "Day";
    public static readonly freshnessWeekText: string = "Week";
    public static readonly freshnessMonthText: string = "Month";

    // Average network timeout in milliseconds.
    public static readonly axiosDefaultTimeout: number = 10000;

    // Leader-board table header key.
    public static readonly leaderBoardTableHeaderKey: string = "header";
}