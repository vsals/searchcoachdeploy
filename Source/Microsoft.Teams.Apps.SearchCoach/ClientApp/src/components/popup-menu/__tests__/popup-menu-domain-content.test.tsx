// <copyright file="popup-menu-domain-content.test.tsx" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import * as React from "react";
import { Provider } from "@fluentui/react-northstar";
import PopUpDomainContent from "../popup-menu-domain-content";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import resources from "../../../constants/resources";
import { ISelectedDomainValue } from "../../../models/ISelectedDomainValue";
import { IDropdownListItem } from "../../../models/IDropdownFilter";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: any) => key,
        i18n: { changeLanguage: jest.fn() },
    }),

    withTranslation: () => (Component: any) => {
        Component.defaultProps = {
            ...Component.defaultProps,
            t: (key: any) => key,
        };
        return Component;
    },
}));

jest.mock("@microsoft/teams-js", () => ({
    initialize: () => {
        return true;
    },
    getContext: (callback: any) =>
        callback(
            Promise.resolve({ teamId: "ewe", entityId: "sdsd", locale: "en-US" })
        ),
}));

let container: any = null;
let initialDomainValuesList: ISelectedDomainValue[] = [];
initialDomainValuesList = resources.domains.map((value: IDropdownListItem) => ({ key: value.id, title: value.name, isChecked: false }));

beforeEach(() => {
    // setup a DOM element as a render target.
    container = document.createElement("div");
    // container *must* be attached to document so events work correctly.
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting.
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("PopUpDomainContent", () => {
    it("renders snapshots", () => {
        act(() => {
            render(
                <Provider>
                    <PopUpDomainContent
                        onDomainValueChange={(selectedDomains: ISelectedDomainValue[]) => { }}
                        selectedDomainValuesList={initialDomainValuesList}
                    />
                </Provider>,
                container
            );
        });

        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});