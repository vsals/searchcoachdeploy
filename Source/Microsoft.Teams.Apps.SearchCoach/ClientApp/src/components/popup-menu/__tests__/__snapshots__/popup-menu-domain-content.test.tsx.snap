// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpDomainContent renders snapshots 1`] = `
"<div class=\\"ui-provider\\" dir=\\"ltr\\" data-uses-unhanded-props=\\"true\\">
  <div class=\\"domain-popup\\">
    <div class=\\"ui-flex domain-container\\" data-uses-unhanded-props=\\"true\\">
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.com</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">comCheckboxLabel</span></div>
      </div>
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.org</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">orgCheckboxLabel</span></div>
      </div>
    </div>
    <div class=\\"ui-flex domain-container\\" data-uses-unhanded-props=\\"true\\">
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.edu</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">eduCheckboxLabel</span></div>
      </div>
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.net</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">netCheckboxLabel</span></div>
      </div>
    </div>
    <div class=\\"ui-flex domain-container\\" data-uses-unhanded-props=\\"true\\">
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.gov</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">govCheckboxLabel</span></div>
      </div>
      <div class=\\"ui-flex domain-button-container\\" data-uses-unhanded-props=\\"true\\">
        <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
          <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
        </div>
        <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">.mil</span><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">milCheckboxLabel</span></div>
      </div>
    </div>
    <div class=\\"ui-flex margin-top\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">domainHelpTitle</span></div>
    <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">domainHelpContent</span></div>
  </div>
</div>"
`;
